src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,62 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+
+public class UserDaoImpl implements UserDao {
+
+    private final SessionFactory sessionFactory;
```

Comment:
Ensure that the `SessionFactory` is properly initialized and injected. If using a dependency injection framework, consider obtaining the `SessionFactory` through it to ensure compatibility and proper lifecycle management.